packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -2058,7 +2064,16 @@ export class MergeTree {
 			}
 			const existingMoveInfo = toMoveInfo(segment);
 
-			if (segment.prevObliterateByInserter?.seq === UnassignedSequenceNumber) {
+			// The "last-to-obliterate-gets-to-insert" policy described by the doc comment on `obliteratePrecedingInsertion`
+			// is mostly handled by logic at insertion time, but we need a small bit of handling here.
+			// Specifically, we want to avoid marking a local-only segment as obliterated when we know one of our own local obliterates
+			// will win against the obliterate we're processing, hence the early exit.
+			if (
```

Comment:
This is the main content of the change, though technically the behavior of how we record `obliteratePrecedingInsertion` is changed as well (note that right here is the only place we meaningfully use it to change any segment state)